internal/commands/app/new_app_inputs.go

```diff
@@ -38,22 +36,21 @@ func (i *newAppInputs) resolveRemoteApp(ui terminal.UI, rc realm.Client) (realm.
 	return ra, nil
 }
 
-func (i *newAppInputs) resolveTemplateID(ui terminal.UI, client realm.Client) error {
-	if i.Template == "" && ui.AutoConfirm() {
+// resolveTemplateID is responsible for resolving a template id from a user cli request.
+// a user can use the --template flag in 1 of the following ways:
+// 1. --template=<template-id>: This populates the value of i.Template to be <template-id> and attempts to create a
+//    template with id = <template-id>
+// 2. no flag defined: This populates the value of i.Template to be "". This tells us to skip the template-app-related
+//    parts of app creation and create a default app
+func (i *newAppInputs) resolveTemplateID(client realm.Client) error {
```

Comment:
I might simplify this docstring a bit (I don't think we need much explanation here about the way flags work/set values)

```go
// resolveTemplateID is responsible for resolving a template id from a user cli request.
// If the --template flag is not set, the CLI will create a default app without any template,
// otherwise the CLI will attempt to create an app based on the specified template
func (i *newAppInputs) resolveTemplateID(client realm.Client) error {
```